### PR TITLE
CA-28793: introduce validation in pool-emergency-reset-master

### PR DESCRIPTION
**Pull request by mseri**
Raise an error and avoid the reset if the ip address is not well-formed or if the host is already the master

Signed-off-by: Marcello Seri <marcello.seri@citrix.com>

### CODE REVIEW

**Comment by mseri** [comment]
Don't merge yet, I want to test it a bit more.

**Comment by robhoes** [comment]
You need to add the new exception to datamodel.ml as well.

File: ocaml/xapi/xapi_pool.ml [hunk 1]
```diff
@@ -800,7 +800,13 @@ let emergency_transition_to_master ~__context =
 let emergency_reset_master ~__context ~master_address =
   if Localdb.get Constants.ha_armed = "true" 
   then raise (Api_errors.Server_error(Api_errors.ha_is_enabled, []));
+  if Pool_role.is_master ()
+  then raise Api_errors.(Server_error(host_is_already_master, []));
```
Comment (robhoes) [inline ocaml/xapi/xapi_pool.ml]: I think that `host_is_master` (from the previous version of the patch) is better. This call is not to make the host a master, but to update a slave's view of where the master is.

**Comment by mseri** [comment]
I have reverted the errors to `host_is_master`. I have also removed the check on ip addresses: when it fails it does it at the `gethostbyname` call (this either fails or resolves partial IPs to local ones, e.g. 127.1 -> 0.0.127.1, and the ip would pass the subsequent validity check)

**Comment by robhoes** [comment]
I'm afraid I had to revert this, because it (unexpectedly) broke pool.join.

**Pull request merged**